### ISSUE TITLE
SSL peer certificate or SSH remote key was not OK

### ISSUE THREAD

**Issue by daftz** [newsboat/newsboat#983]
newsboat 2.19.0
System: Darwin 19.5.0 (x86_64)
Compiler: g++ 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.17)
ncurses: ncurses 5.7.20081102 (compiled with 5.7)
libcurl: libcurl/7.64.1 (SecureTransport) LibreSSL/2.8.3 zlib/1.2.11 nghttp2/1.39.2 (compiled with 7.64.1)
SQLite: 3.32.1 (compiled with 3.31.1)
libxml2: compiled with 2.9.4

Config file (copy from ~/.newsboat/config or ~/.config/newsboat/config):

```
# -- feeds ---------------------------------------------------------------------

#auto-reload yes
#reload-time 120
reload-threads 4
#download-retries 4
#download-timeout 10
prepopulate-query-feeds yes
download-path "~/Downloads/%h/%n"

# -- display -------------------------------------------------------------------

show-read-articles no
show-read-feeds no
prepopulate-query-feeds yes
feed-sort-order unreadarticlecount
show-keymap-hint yes
swap-title-and-hints no
text-width 72
save-path "~/Documents/RSS/
#browser "open /Applications/Firefox.app"  %u
browser lynx %u
confirm-exit yes
display-article-progress yes

feed-sort-order title

#bind-key b open /Applications/Firefox.app %u
bind-key U show-urls

macro f set browser "open /Applications/Firefox.app %u"; open-in-browser ; set browser lynx
macro , open-in-browser

#color info default default reverse
#color listnormal_unread yellow default
#color listfocus blue default reverse bold
#color listfocus_unread blue default reverse bold

color listnormal blue default
color listfocus black yellow standout bold
color listnormal_unread white default
color listfocus_unread yellow default bold
color info red black bold
color article green default
```

Steps to reproduce the issue:

1. Subscribe to url: https://www.rtvutrecht.nl/rss/nieuws.xml

2. When refreshing this feed, the error "SSL peer certificate or SSH remote key was not OK".shows up.

I checked the certificate of the site and it's valid. Testing the feed in Vienna gives no problems.

Please let me know if you need more info.

Thank you!

**Other info you think is relevant**:

**Comment by Minoru** [comment]
Hi! I don't have LibreSSL, but I reproduced this with GnuTLS 3.6.13. If I link Newsboat against OpenSSL 1.1.1g, it fetches the feed fine. This indicates that the problem *is* with the certificate.

Another user on IRC reported the same error message produced with GnuTLS and https://goinglinux.com/oggpodcast.xml.

If you look at certification paths in [SSLLab's report for rtvutrecht.nl](https://www.ssllabs.com/ssltest/analyze.html?d=www.rtvutrecht.nl&s=2a02%3a458%3a101%3a28%3a100%3a28%3a0%3ad9&hideResults=on), you'll see there are multiple paths, and one of them isn't trusted because [AddTrust root expired at the end of May 2020](https://support.sectigo.com/articles/Knowledge/Sectigo-AddTrust-External-CA-Root-Expiring-May-30-2020). Apparently OpenSSL (and Vienna) take one (trusted) path, but GnuTLS (and LibreSSL) take another (untrusted) path. (Same thing with happened with goinglinux.com.)

You can disable path verification with `ssl-verifypeer no`, but that'll affect *all* your feeds. Consider emailing the site admin to notify them of the problem. I believe all they need to do is stop sending an expired root certificate as part of the SSL handshake.

Since this is not a problem with Newsboat per se, I'm tagging this bug as invalid and closing the issue. Thanks for taking the time to write a high-quality report, though; it definitely sped up the debugging on my end!

**Comment by Minoru** [comment]
Debian testing just shipped ca-certificates 20200601, which [removes the aforementioned expired AddTrust Root](https://metadata.ftp-master.debian.org/changelogs//main/c/ca-certificates/ca-certificates_20200601_changelog). Now I can't reproduce this bug even with GnuTLS. So it looks like you can work around this by updating your certificates store. Note that Newsboat respects `CURL_CA_BUNDLE` environment variable, so you can probably take your system's bundle, remove the offending certificate, and instruct Newsboat (and curl) to use it.

Also saved [current SSLLabs report for rtvutrecht.nl on the Wayback Machine]( https://web.archive.org/web/20200607132049/https://www.ssllabs.com/ssltest/analyze.html?d=www.rtvutrecht.nl&s=2a02%3A458%3A101%3A28%3A100%3A28%3A0%3Ad9&hideResults=on).

**Comment by daftz** [comment]
Thanks for the followup! It seems that the site has fixed the problem themselves, the feed is working again :)
